### PR TITLE
Refactored package

### PR DESCRIPTION
**Pull request by makhosravi**
The core feature dart file has the name of the feature.
All model classes are in the ```model.dart``` file.
Constants moved to the ```lib/src/constants/feature_name/constants.dart``` file.
Values that are not used as public APIs were converted to private access.

### CODE REVIEW

File: lib/src/core/bill/bill.dart [hunk 1]
```diff
@@ -46,16 +35,15 @@ class Bill {
...
   }
 
   /// Takes bill type of the Bill form bill ID
   String get billType {
     var billIdStr = '$_billId';
-    var billTypeKey = int.parse(
-        billIdStr.substring(billIdStr.length - 2, billIdStr.length - 1));
-    return billTypesMap[billTypeKey] ?? 'unknown';
+    var billTypeKey = int.parse(billIdStr.substring(billIdStr.length - 2, billIdStr.length - 1));
+    return billTypesMap[billTypeKey] ?? unknown;
```
Comment (payam-zahedi) [inline lib/src/core/bill/bill.dart:46]: Are you sure that you run the `dart format` command?

File: lib/src/core/vehicle_plate/vehicle_plate.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+abstract class _PlateDataset {
+  /// Returns car province name by the given key which is province code
+  static String? getCarProvince(int key) => plateDataset[0][key];
+
+  /// Returns motorcycle province name by the given key which is province code
+  static String? getMotorcycleProvince(int key) => plateDataset[1][key];
+
+  /// Returns category name by the given char
+  static String? getCategory(String key) => plateDataset[2][key];
+}
```
Comment (payam-zahedi) [inline lib/src/core/vehicle_plate/vehicle_plate.dart:14]: Why should we have an abstract class here? We can turn its methods into the high-order private methods.

**Review by payam-zahedi — CHANGES_REQUESTED** [review]
Excellent work @makhosravi. 
Before merging this partnership, you should consider the following tasks.

- run `dart test .`  command
- run dart format command
- Make sure commit messages are up to 70 characters long.
- make sure that you updated the README.md file

**Comment by payam-zahedi** [comment]
@EhsanAramide I think you should also review this PR.

**Review by EhsanAramide — COMMENTED** [review]
Well done.
I am very interested about your contribution but before merging first of all check the testcases in `test` directory then check the `example` directory and ensure everything works fine.
Also take code coverage after refactoring it shouldn't be under 95 percent. (Instructions are available in `.github/workflows/test.yml` directory )

**Comment by ali-master** [comment]
Well done boys, you are all heroes. Thanks for helping the PersianTools family to grow faster.

**Review by payam-zahedi — APPROVED** [review]
Well done Khosro, everything looks great.

**Comment by payam-zahedi** [comment]
@EhsanAramide can you check the changes too? I don't see any problem

**Review by EhsanAramide — COMMENTED** [review]
Well done, everything was fine.

**Pull request merged**